docker/drone/src/main/java/org/arquillian/cube/docker/drone/VncRecorderLifecycleManager.java

```diff
@@ -0,0 +1,124 @@
+
+        if (cubeDroneConfiguration.isRecording()) {
+
+            // lazy init
+            initVncCube(cubeRegistry);
+
+            vnc.create();
+            vnc.start();
+
+        }
+
+    }
+
+    private void initVncCube(CubeRegistry cubeRegistry) {
+        if (vnc == null) {
+            Cube vncContainer = cubeRegistry.getCube(SeleniumContainers.VNC_CONTAINER_NAME);
+
+            if (vncContainer == null) {
+                throw new IllegalArgumentException("VNC cube is not present in registry.");
+            }
+
+            this.vnc = vncContainer;
+        }
+    }
+
+    public void stopRecording(@Observes After afterTestMethod, TestResult testResult, CubeDroneConfiguration cubeDroneConfiguration, SeleniumContainers seleniumContainers) {
+
+        if (this.vnc != null) {
+
+            if (cubeDroneConfiguration.isRecordingOnlyFailing() && testResult.getStatus() == TestResult.Status.FAILED) {
+                moveFromVolumeFolderToBuildDirectory(afterTestMethod, cubeDroneConfiguration, seleniumContainers);
+            } else {
+                // RECORD ALWAYS
+                if (cubeDroneConfiguration.isRecording() && !cubeDroneConfiguration.isRecordingOnlyFailing()) {
+                    moveFromVolumeFolderToBuildDirectory(afterTestMethod, cubeDroneConfiguration, seleniumContainers);
+                }
+            }
+
+            vnc.stop();
+            vnc.destroy();
+        }
+
+    }
+
+    private void moveFromVolumeFolderToBuildDirectory(After afterTestMethod, CubeDroneConfiguration cubeDroneConfiguration, SeleniumContainers seleniumContainers) {
+        try {
+            Files.move(seleniumContainers.getVideoRecordingFile(), getFinalLocation(cubeDroneConfiguration, afterTestMethod), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private Path getFinalLocation(CubeDroneConfiguration cubeDroneConfiguration, After afterTestMethod) {
+        return resolveTargetDirectory(cubeDroneConfiguration).resolve(getFinalVideoName(afterTestMethod));
+    }
+
+    private Path resolveTargetDirectory(CubeDroneConfiguration cubeDroneConfiguration) {
+        if (cubeDroneConfiguration.isVideoOutputDirectorySet()) {
+            return Paths.get(cubeDroneConfiguration.getFinalDirectory());
+        } else {
+            if (MAVEN_REPORT_DIR.exists()) {
+                return MAVEN_REPORT_DIR.toPath();
+            } else {
+                if (GRADLE_REPORT_DIR.exists()) {
+                    return GRADLE_REPORT_DIR.toPath();
+                } else {
+                    if (DEFAULT_LOCATION_OUTPUT_GRADLE.exists()) {
+                        return DEFAULT_LOCATION_OUTPUT_GRADLE.toPath();
+                    } else {
+                        if (DEFAULT_LOCATION_OUTPUT_MAVEN.exists()) {
+                            return DEFAULT_LOCATION_OUTPUT_MAVEN.toPath();
+                        }
```

Comment:
Any idea that it is easy to follow and fast to execute? I thought on Chain Pattern but maybe it is an overkill for just 5 `ifs`without any logic rather than returning a string.